src/calendar/view/CalendarAgendaViewModel.ts

```diff
@@ -0,0 +1,20 @@
+import { CalendarEvent } from "../../api/entities/tutanota/TypeRefs.js"
+import { isAllDayEvent } from "../../api/common/utils/CommonCalendarUtils.js"
+
+export class CalendarAgendaViewModel {
+	earliestEventToShowTimeIndicator(events: readonly CalendarEvent[], date: Date): number | null {
```

Comment:
I think it is small enough to just be a function that we can test, ViewModels usually have state/dependencies and we already have one for calendar.